posts/2021-02-11-Rust-1.50.0.md

```diff
@@ -0,0 +1,170 @@
+---
+layout: post
+title: "Announcing Rust 1.50.0"
+author: The Rust Release Team
+release: true
+---
+
+The Rust team is happy to announce a new version of Rust, 1.50.0. Rust is a
+programming language that is empowering everyone to build reliable and
+efficient software.
+
+If you have a previous version of Rust installed via rustup, getting Rust
+1.50.0 is as easy as:
+
+```console
+rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`][install]
+from the appropriate page on our website, and check out the
+[detailed release notes for 1.50.0][notes] on GitHub.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1500-2021-02-11
+
+## What's in 1.50.0 stable
+
+For this release, we have improved array indexing, expanded safe access to union fields, and added to the standard library.
+See the [detailed release notes][notes] to learn about other changes
+not covered by this post.
+
+### Const-generic array indexing
+
+Continuing the march toward stable `const` generics, this release adds
+implementations of `ops::Index` and `IndexMut` for arrays `[T; N]` for
+_any_ length of `const N`. The indexing operator `[]` already worked on
+arrays through built-in compiler magic, but at the type level, arrays
+didn't actually implement the library traits until now.
+
+```rust
+fn second<C>(container: &C) -> &C::Output
+where
+    C: std::ops::Index<usize> + ?Sized,
+{
+    &container[1]
+}
+
+fn main() {
+    let array: [i32; 3] = [1, 2, 3];
+    assert_eq!(second(&array[..]), &2); // slices worked before
+    assert_eq!(second(&array), &2); // now it also works directly
+}
+```
+
+### `const` value repetition for arrays
+
+Arrays in Rust can be written either as a list `[a, b, c]` or a repetition `[x; N]`.
+For lengths `N` greater than one, repetition has only been allowed for `x`s that are `Copy`,
+and [RFC 2203] sought to allow any `const` expression there. However,
+while that feature was unstable for arbitrary expressions, its implementation
+since Rust 1.38 accidentally allowed stable use of `const` _values_ in array
+repetition.
+
+```rust
+fn main() {
+    // This is not allowed, because `Option<Vec<i32>>` does not implement `Copy`.
+    let array: [Option<Vec<i32>>; 10] = [None; 10];
+
+    const NONE: Option<Vec<i32>> = None;
+    const EMPTY: Option<Vec<i32>> = Some(Vec::new());
+
+    // However, repeating a `const` value is allowed!
+    let nones = [NONE; 10];
+    let empties = [EMPTY; 10];
+}
+```
+
+In Rust 1.50, that stabilization is formally acknowledged, so you may use this
+construct without fear. In the future, to avoid such "temporary" named
```

Comment:
Hm, I'd prefer to avoid this "without fear" language - I think it's unlikely we would have dropped support for such a long-term stabilized feature regardless, and it indicates some worry about backwards compatibility in my eyes.